### ISSUE TITLE
PHP 8.0 / PHP 7.4

### ISSUE THREAD

**Issue by Domgarth** [iCalcreator/iCalcreator#108]
I see below there are 3 builds of the lib
Is there anyway to support both php 8.0 and php 7.4 so that the same build of our website supports both versions?
Does v2.42 support this? Or could the correct version be loaded depending on php 7.4 or php 8.0?

v2.41.74 - v2.42 pre-release
v2.40 - stable PHP8
v2.39 - PHP7

**Comment by iCalcreator** [comment]
The 2.41.* releases support PHP 8,
a downgrade to 7.4 (i.e a new 2.39.x release) requires some substantial sponsorship...

**Comment by phpipam** [comment]
Hi, first thanks for good work :)

I get this deprecation error, might be good to fix for next php releases:
```
Deprecated:  Creation of dynamic property DateInterval::$days is deprecated in /var/www/html/assets/composer/vendor/kigkonsult/icalcreator/src/Util/DateIntervalFactory.php on line 390
Deprecated:  Creation of dynamic property DateInterval::$from_string is deprecated in /var/www/html/assets/composer/vendor/kigkonsult/icalcreator/src/Util/DateIntervalFactory.php on line 390
```

Version:
```
kigkonsult/icalcreator        v2.41.56 iCalcreator is the PHP implementation of rfc2445/rfc5545 and rfc updates, management of calendar information
```

php version:
```
PHP 8.3.2 (cli) (built: Jan 16 2024 13:46:41) (NTS)
```

**Comment by iCalcreator** [comment]
Fixed in the current version, 2.41.90, please update and check again!
